### ISSUE TITLE
No solution for BST Delete Node & missing root

### ISSUE THREAD

**Issue by nicholaai** [kuychaco/algoClass#11]
Hi, I've made my way here from front end masters. Your course has been great!

I've come to the BST section of your course and in the videos you go over the solutions to deleting a node from a binary search tree. Issues are:

- You only write pseudo code for the cases that have 0 or 1 child nodes. No solution is shown in the videos or within your solution branches.

- For the 3rd case in which there are 2 children nodes, you show the code for when we are trying to delete the root and when the value we are deleting is not the root. Issue is, you add in a root property to the code, when throughout the rest of the BST constructor we have not used a root. How would this affect the rest of the code? Would we just replace our parent checks with referencing the root? 

Essentially I'm asking for the BST solution to be updated to a full solution. Thank you.

**Comment by invalidred** [comment]
I too am here from front end masters and truly enjoyed the course.

I too have the same request as nicholaai.

**Comment by pemigra** [comment]
Yes, any chance we can see the full solution for deleting a node in a BST?
